services/venice-controller/src/main/java/com/linkedin/venice/controller/kafka/TopicCleanupService.java

```diff
@@ -200,28 +221,70 @@ void cleanupVeniceTopics() {
   }
 
   private void populateDeprecatedTopicQueue(PriorityQueue<PubSubTopic> topics) {
-    Map<String, Map<PubSubTopic, Long>> allStoreTopics = getAllVeniceStoreTopicsRetentions(getTopicManager());
+    Map<PubSubTopic, Long> topicsWithRetention = getTopicManager().getAllTopicRetentions();
+    if (!multiDataCenterStoreToVersionTopicCount.isEmpty()) {
+      refreshMultiDataCenterStoreToVersionTopicCountMap(topicsWithRetention.keySet());
+    }
+    Map<String, Map<PubSubTopic, Long>> allStoreTopics = getAllVeniceStoreTopicsRetentions(topicsWithRetention);
     allStoreTopics.forEach((storeName, topicRetentions) -> {
+      int minNumOfUnusedVersionTopicsOverride = minNumberOfUnusedKafkaTopicsToPreserve;
       PubSubTopic realTimeTopic = pubSubTopicRepository.getTopic(Version.composeRealTimeTopic(storeName));
       if (topicRetentions.containsKey(realTimeTopic)) {
         if (admin.isTopicTruncatedBasedOnRetention(topicRetentions.get(realTimeTopic))) {
           topics.offer(realTimeTopic);
+          minNumOfUnusedVersionTopicsOverride = 0;
         }
         topicRetentions.remove(realTimeTopic);
       }
       List<PubSubTopic> oldTopicsToDelete =
-          extractVersionTopicsToCleanup(admin, topicRetentions, minNumberOfUnusedKafkaTopicsToPreserve, delayFactor);
+          extractVersionTopicsToCleanup(admin, topicRetentions, minNumOfUnusedVersionTopicsOverride, delayFactor);
       if (!oldTopicsToDelete.isEmpty()) {
         topics.addAll(oldTopicsToDelete);
       }
     });
   }
 
+  private void refreshMultiDataCenterStoreToVersionTopicCountMap(Set<PubSubTopic> localTopics) {
+    clearAndPopulateStoreToVersionTopicCountMap(localTopics, multiDataCenterStoreToVersionTopicCount.get(0));
+    if (childFabricList.size() > 1) {
+      if (localPubSubBootstrapServer == null) {
+        localPubSubBootstrapServer = getTopicManager().getPubSubBootstrapServers();
+      }
+      int i = 1;
+      for (String childFabric: childFabricList) {
+        // Best effort approach when fetching topics from other fabrics
+        try {
+          String pubSubBootstrapServer = multiClusterConfigs.getChildDataCenterKafkaUrlMap().get(childFabric);
+          if (localPubSubBootstrapServer.equals(pubSubBootstrapServer)) {
+            continue;
+          }
+          Set<PubSubTopic> remoteTopics = getTopicManager(pubSubBootstrapServer).getAllTopicRetentions().keySet();
+          clearAndPopulateStoreToVersionTopicCountMap(remoteTopics, multiDataCenterStoreToVersionTopicCount.get(i));
+          i++;
+        } catch (Exception e) {
+          LOGGER.error("Failed to refresh store to version topic count map for fabric {}", childFabric, e);
+        }
```

Comment:
Will block RT topic deletion if we:

1. Unable to make sense of the child fabric list or kafka URLs (mis-configured)
2. Unable to fetch version topic info because kafka is down/unreachable like you mentioned in the above scenario.